src/Utilities/InfoStream.h

```diff
@@ -0,0 +1,93 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2017 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/** @file InfoStream.h
+ * @brief Declaration of InfoStream class.
+ */
+
+#ifndef INFOSTREAM_H
+#define INFOSTREAM_H
+
+#include <string>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+
+// Bit bucket to discard output
+// From https://stackoverflow.com/questions/7818371/printing-to-nowhere-with-ostream
+class NullStreamBuf : public std::streambuf
+{
+protected:
+    virtual int overflow(int c) { return c; }
+};
+
+class NullStream : public NullStreamBuf, public std::ostream
```

Comment:
In the old code, we have std::ostream(0) as a null stream. This was because std::ostream() is not thread safe and crashing the code. I feel that we need to use the old way since NullStream is not safe.